### PR TITLE
added movies search

### PR DESCRIPTION
**Pull request by ElenaLadutska**

### CODE REVIEW

File: my-app/src/App.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+import './App.css';
+import Movies from './GetMovies';
+
+function App() {
+  return (
+    <>
+      <Movies />
+    </>
```
Comment (kathrynowy) [inline my-app/src/App.js]: do you need a fragment here?

File: my-app/src/GetMovies.jsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import { useEffect, useState } from 'react';
+
+function Movies() {
+  const [movies, setMovies] = useState([]);
+  const [value, setValue] = useState('');
```
Comment (kathrynowy) [inline my-app/src/GetMovies.jsx]:
```suggestion
  const [movies, setMovies] = useState([]);
  
  const [value, setValue] = useState('');
```

File: my-app/src/GetMovies.jsx [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+  const getMovies = async() => {
+    fetch("http://localhost:3000/movies.json")
+    .then(response => response.json())
+    .then(data => setMovies(data))
```
Comment (kathrynowy) [inline my-app/src/GetMovies.jsx]: try/catch

File: my-app/src/GetMovies.jsx [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  };
+
+  useEffect(() => {
+    getMovies();
+  }, [value]);
```
Comment (kathrynowy) [inline my-app/src/GetMovies.jsx]: ? if you don't use value - you don't need to pass variable into the dependencies array

File: my-app/src/GetMovies.jsx [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+  const filteredMovies = movies.filter(movie => {
+    return movie.name.toLowerCase().includes(value.toLowerCase())
+  })
```
Comment (kathrynowy) [inline my-app/src/GetMovies.jsx]: 1 use callback и прочитать зачем он нам нужен
2 убрать ретурн

File: my-app/src/GetMovies.jsx [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+      placeholder="Enter movies title..."/>
+
+      <ul className="app">
+        {movies &&
+          filteredMovies.map((movie, index) => (
+            <li key={index}>{movie.name}</li>
+        ))
+        }
+      </ul>
+    </>
```
Comment (kathrynowy) [inline my-app/src/GetMovies.jsx]: .

**Review by kathrynowy — CHANGES_REQUESTED** [review]

File: my-app/src/App.js (hunk 1 above)
Comment (ElenaLadutska) [reply my-app/src/App.js]: no, removed

File: my-app/src/GetMovies.jsx (hunk 2 above)
Comment (ElenaLadutska) [reply my-app/src/GetMovies.jsx]: fixed

File: my-app/src/GetMovies.jsx (hunk 4 above)
Comment (ElenaLadutska) [reply my-app/src/GetMovies.jsx]: okay, thanks, I removed value from the dependencies array

File: my-app/src/GetMovies.jsx (hunk 6 above)
Comment (ElenaLadutska) [reply my-app/src/GetMovies.jsx]: fixed

File: my-app/src/GetMovies.jsx (hunk 5 above)
Comment (ElenaLadutska) [reply my-app/src/GetMovies.jsx]: fixed

File: my-app/src/App.js [hunk 7]
```diff
@@ -1,13 +1,25 @@
 import React from 'react';
-import './App.css';
-import Movies from './GetMovies';
+import { Routes, Route } from 'react-router-dom';
+import { useState } from "react";
+import './styles/App.css';
+import Movies from './Movies';
+import Movie from './components/Movie';
+import Layout from './components/Layout';
+import NotFound from './components/NotFound';
```
Comment (kathrynowy) [inline my-app/src/App.js:8]: import orders

File: my-app/src/App.js [hunk 8]
```diff
@@ -1,13 +1,25 @@
...
+const App = () => {
+  const [city, setCity] = useState("1");
+
+  return(
+  <>
+    <Routes>
+      <Route path='/' element={<Layout city={city} setCity={setCity}/>}>
+        <Route index element={<Movies city={city}/>} />
+        <Route path='movie-description/:id' element={<Movie city={city}/>} />
+        <Route path='*' element={<NotFound />}/>
```
Comment (kathrynowy) [inline my-app/src/App.js]: map

File: my-app/src/App.js [hunk 9]
```diff
@@ -1,13 +1,25 @@
...
 
-function App() {
-  return (
-    <>
-      <Movies />
-    </>
-  );
-}
+const App = () => {
+  const [city, setCity] = useState("1");
```
Comment (kathrynowy) [inline my-app/src/App.js:15]: why do you need it?

File: my-app/src/Movies.jsx [hunk 10]
```diff
@@ -0,0 +1,115 @@
+import { useEffect, useState, useCallback } from "react";
+import { Link } from "react-router-dom";
+import Button from "./components/Button";
+import Loader from "./components/Loader";
+import ScrollButton from "./components/ScrollButton";
+
+function Movies(props) {
+  let [isLoading, setLoading] = useState(true);
```
Comment (kathrynowy) [inline my-app/src/Movies.jsx:9]: why let??

Comment (kathrynowy) [reply my-app/src/Movies.jsx:9]: why isLoading true ?

isLoading only when you start requesting data

File: my-app/src/Movies.jsx [hunk 11]
```diff
@@ -0,0 +1,115 @@
...
+
+  let [isError, setError] = useState(false);
+
+  const getMovies = async() => {
+    try {
+      fetch(`https://soft.silverscreen.by:8443/wssite/webapi/event/data?filter=%7B%22city%22:${props.city}%7D&extended=true`)
+        .then(response => response.json())
+        .then(data => setMovies(data));
+
+      setLoading(isLoading = false);
```
Comment (kathrynowy) [inline my-app/src/Movies.jsx]: (((( 
read docs about usestate pls

Comment (kathrynowy) [reply my-app/src/Movies.jsx]: for every single useState you have written, it's all wrong

File: my-app/src/Movies.jsx [hunk 12]
```diff
@@ -0,0 +1,115 @@
...
+    } 
+    catch (error) {
+      setError(isError = true);
+      throw new Error("Something wrong with server")
+    }
+  };
+
+  useEffect(() => {
+    getMovies();
```
Comment (kathrynowy) [inline my-app/src/Movies.jsx]: getMovies should be in the dependency array

File: my-app/src/Movies.jsx [hunk 13]
```diff
@@ -0,0 +1,115 @@
...
+  }, []);
+
+  const [filteredMovies, setFilteredMovies] = useState([]);
```
Comment (kathrynowy) [inline my-app/src/Movies.jsx]: at first all useStates
than functionss / useCallbacks

than useEffects

File: my-app/src/Movies.jsx [hunk 14]
```diff
@@ -0,0 +1,115 @@
...
+
+  const filterMovies = useCallback(
+    () => {
+      setFilteredMovies(movies.filter(movie => movie.name.toLowerCase().includes(value.toLowerCase())))
```
Comment (kathrynowy) [inline my-app/src/Movies.jsx]: too long line

Comment (kathrynowy) [reply my-app/src/Movies.jsx]: you can destructure a few fields here

File: my-app/src/Movies.jsx [hunk 15]
```diff
@@ -0,0 +1,115 @@
...
+    },
+    [movies, value],
+  );
+
+  useEffect(() => {
+    filterMovies()
```
Comment (kathrynowy) [inline my-app/src/Movies.jsx:49]: use useMemo for filtering movies pls

File: my-app/src/Movies.jsx [hunk 16]
```diff
@@ -0,0 +1,115 @@
...
+
+    window.addEventListener('scroll', handleScroll);
+
+    return () => {
+      window.removeEventListener('scroll', handleScroll);
+    };
+  }, []);
+
+  return (
+    isLoading ? <Loader /> : (
```
Comment (kathrynowy) [inline my-app/src/Movies.jsx]: move above to if statement pls
it looks very bad, unreadable

File: my-app/src/Movies.jsx [hunk 17]
```diff
@@ -0,0 +1,115 @@
...
+                    id={index} 
+                    title="Buy a ticket"
+                />
+              </Link>
+              </li>
+          ))
+          }
+
+          {
+            !!movies.length || (
```
Comment (kathrynowy) [inline my-app/src/Movies.jsx]: if we have movies we render we don't have them?

File: my-app/src/Movies.jsx [hunk 18]
```diff
@@ -0,0 +1,115 @@
...
+          {
+            isError && (
+              <div className="error">Oops, something went wrong</div>
+            )
+          }
+        </ul>
+
+        {
+          scrollTop > 200 ? <ScrollButton /> : ""
+        }
```
Comment (kathrynowy) [inline my-app/src/Movies.jsx]: what is ''???

File: my-app/src/Movies.jsx [hunk 19]
```diff
@@ -0,0 +1,115 @@
...
+        <input 
+          type="text" 
+          onChange={event => setValue(event.target.value)}
+          value={value}
+          placeholder="Enter movies title..."
+        />
+
+        <ul className="movies">
+          {!!movies?.length &&
+            filteredMovies.map((movie, index) => (
```
Comment (kathrynowy) [inline my-app/src/Movies.jsx]: destructuring

File: my-app/src/Movies.jsx [hunk 20]
```diff
@@ -0,0 +1,115 @@
...
+                </p>
+
+              <Link to={`/movie-description/${movie.eventId}`}>
+                <Button 
+                    id={index} 
+                    title="Buy a ticket"
+                />
+              </Link>
+              </li>
+          ))
```
Comment (kathrynowy) [inline my-app/src/Movies.jsx]: one movie is a component

File: my-app/src/Movies.jsx [hunk 21]
```diff
@@ -0,0 +1,115 @@
...
+                <img 
+                  alt="movieImg"
+                  src={movie.posterLink}>
+                </img>
+
+                <p>
+                  {movie.name}
+                </p>
+
+              <Link to={`/movie-description/${movie.eventId}`}>
```
Comment (kathrynowy) [inline my-app/src/Movies.jsx]: i've told this while meeting
we use handler function with router push
not Link

File: my-app/src/components/Button/index.jsx [hunk 22]
```diff
@@ -0,0 +1,9 @@
+function Button(props) {
+  return (
+    <button className={props.className} id={props.id} > 
+      {props.title}
```
Comment (kathrynowy) [inline my-app/src/components/Button/index.jsx]: 1. arrow functions
2. destructuring
3 why do you need id here for the btn?

File: my-app/src/components/Cinemas/index.jsx [hunk 23]
```diff
@@ -0,0 +1,47 @@
...
+    return movieDate
+  }
+
+  const startTime = (place) => {
+    const time = new DateObject(place).format("HH:mm");
+
+    return time;
+  };
+
+  return (
```
Comment (kathrynowy) [inline my-app/src/components/Cinemas/index.jsx]: .
services 
+ remove unnecessary variable

File: my-app/src/components/Cinemas/index.jsx [hunk 24]
```diff
@@ -0,0 +1,47 @@
...
+    <>
+      {name.film.showList
+        ? Object.entries(name.film.showList).map(([date, value], index) => (
```
Comment (kathrynowy) [inline my-app/src/components/Cinemas/index.jsx]: в беседе

File: my-app/src/components/Cinemas/index.jsx [hunk 25]
```diff
@@ -0,0 +1,47 @@
...
+          <div 
+            key={index}
+            className="showDate"
```
Comment (kathrynowy) [inline my-app/src/components/Cinemas/index.jsx]: naming

File: my-app/src/components/Cinemas/index.jsx [hunk 26]
```diff
@@ -0,0 +1,47 @@
...
+                  key={index}>
+                    
+                    <div className="place">{place.theater.name}</div>
+
+                    <div className="time">
+                      {startTime(Date.parse(place.start))}
+                    </div>
+                </div>
+              )})}
+          </div>
```
Comment (kathrynowy) [inline my-app/src/components/Cinemas/index.jsx]: chat tg

File: my-app/src/styles/index.css [hunk 27]
```diff
@@ -0,0 +1,230 @@
...
+#cities option {
+  background-color: #000;
+  color: #fff;
+}
+
+#cities:focus {
+  outline: none;
+  border: 1px solid #0ebeac;
+}
+
```
Comment (kathrynowy) [inline my-app/src/styles/index.css:378]: move all styles to appropriate filders

**Review by kathrynowy — CHANGES_REQUESTED** [review]

**Pull request merged**